### PR TITLE
[ci][py11] support docker image build for x86

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
Support Linux cpu docker image build for py11

## Checks
- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - [X] Unit tests

### CODE REVIEW

File: .buildkite/pipeline.build.yml
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:67]: this is `cpu/cu115/cu116` I think.

File: docker/base-deps/Dockerfile
Comment (aslonnie) [inline docker/base-deps/Dockerfile:64]: I think it was intentionally using one RUN to keep it as one docker layer.

If you need more fancy/complicated stuff, maybe start using `DOCKER_BUILDKIT=1` and use a full bash script.

File: python/requirements/ml/requirements_no_deps.txt
Comment (aslonnie) [inline python/requirements/ml/requirements_no_deps.txt:6]: what exactly changed here?

File: docker/base-deps/Dockerfile
Comment (can-anyscale) [reply docker/base-deps/Dockerfile:51]: i like that

File: python/requirements/ml/requirements_no_deps.txt
Comment (can-anyscale) [reply python/requirements/ml/requirements_no_deps.txt:6]: supposedly no change, let me see if i can get rid of this

File: docker/base-deps/Dockerfile
Comment (can-anyscale) [reply docker/base-deps/Dockerfile]: I used RUN << EOF without success, so I inline everything into one RUN call. My guess is DockerClient python API does not understand that format.

File: docker/base-deps/Dockerfile
Comment (rickyyx) [inline docker/base-deps/Dockerfile:54]: Could we add a TODO for the related libffi=3.3 patch? I remember there's a link to an issue on conda.

Comment (rickyyx) [reply docker/base-deps/Dockerfile:54]: https://github.com/ray-project/ray/issues/33299

**Review by rickyyx — APPROVED** [review]
LGTM! 

But I will wait for opinions from the CI team for a better stamp.

File: docker/base-deps/Dockerfile
Comment (can-anyscale) [reply docker/base-deps/Dockerfile:54]: Totally!

**Comment by can-anyscale** [comment]
I think I still need an approval from @aslonnie or @krfricke as a code owner. Thankkks

File: docker/base-deps/Dockerfile
Comment (aslonnie) [inline docker/base-deps/Dockerfile:50]: nit: use `if [[ "PYTHON_VERSION" == "3.11" ]]; then`

also `PYTHON_VERSION` seems to have 3 number, as on line 14:

`PYTHON_VERSION=3.8.16`

so maybe you want to use `if [[ "${PYTHON_VERSION}" == "3.11" || "${PYTHON_VERSION}" =~ ^3\.11\. ]];`


or change line 14..

File: docker/base-deps/Dockerfile
Comment (aslonnie) [inline docker/base-deps/Dockerfile:51]: please use an array:

`LIBGCC=(libgcc-ng)` here.

File: docker/base-deps/Dockerfile
Comment (aslonnie) [inline docker/base-deps/Dockerfile:55]: and `LIBGCC=(libgcc-ng libffi=3.3)` here.

File: docker/base-deps/Dockerfile
Comment (aslonnie) [inline docker/base-deps/Dockerfile:64]: and for this line, use `${LIBGCC[@]}" to expand the array.

File: docker/base-deps/Dockerfile
Comment (can-anyscale) [reply docker/base-deps/Dockerfile]: Lol I tried this and [[ ]], both are rejected by Docker for some reason. E.g. this is the error for using (): https://buildkite.com/ray-project/oss-ci-build-pr/builds/26300#0188e022-f8ab-4847-b721-db4c8cdf19ad. 

Going back to using [] and string seems to be ok.

Maybe I poke you tomorrow on the syntax to see what's wrong

**Review by aslonnie — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/FarCryGame/">Far Cry 6</a> on <a href="https://giphy.com/gifs/FarCryGame-crying-richard-far-cry-6-QAJGv5Y9XOjS8OOrtd">GIPHY</a></div>

**Review by krfricke — APPROVED** [review]

**Pull request merged**
